### ISSUE TITLE
Multi-threaded execution is actually slower (checked on DKI model only)

### ISSUE THREAD

**Issue by oesteban** [nipreps/eddymotion#101]
The most striking issue is how useless (actually detrimental) is fitting in parallel:

Single thread:
```
Model fit (218370 voxels, 256 orientations) - 137.28477546899921 s.
```

Multi-thread (16 threads):
```
Model fit (13648 voxels, 256 orientations) - 172.66635700699953 s.
Model fit (13648 voxels, 256 orientations) - 173.20241945700036 s.
Model fit (13648 voxels, 256 orientations) - 173.40447643799962 s.
Model fit (13648 voxels, 256 orientations) - 174.46670007600005 s.
Model fit (13649 voxels, 256 orientations) - 174.56820751199848 s.
Model fit (13648 voxels, 256 orientations) - 174.708012770001 s.
Model fit (13648 voxels, 256 orientations) - 175.30970972500108 s.
Model fit (13648 voxels, 256 orientations) - 175.30730865499936 s.
Model fit (13648 voxels, 256 orientations) - 175.5713644539992 s.
Model fit (13648 voxels, 256 orientations) - 175.6265785329997 s.
Model fit (13648 voxels, 256 orientations) - 175.94506397100122 s.
Model fit (13648 voxels, 256 orientations) - 176.01709744599975 s.
Model fit (13648 voxels, 256 orientations) - 176.1737611839999 s.
Model fit (13649 voxels, 256 orientations) - 176.2048280279996 s.
Model fit (13648 voxels, 256 orientations) - 176.2146118239998 s.
Model fit (13648 voxels, 256 orientations) - 176.2184523369997 s.
```

@arokem any suggestions of why DKI.fit_model takes about the same time (or even more) with 1/16 of the voxels?

**Comment by arokem** [comment]
Might be a speed/memory tradeoff. DKI and DTI are both implemented to use multi-threaded array operations in numpy already: https://github.com/dipy/dipy/blob/master/dipy/reconst/dti.py#L1397 (see also: https://github.com/dipy/dipy/blob/d0401e0d8f4234997f2a85a70782ddec005b12b5/dipy/reconst/dti.py#L1226), so I don't think that you will get much of a speedup for these models through this parallelization. Can you please try this with something like SFM?

**Comment by oesteban** [comment]
That's what I thought, numpy's parallelization is definitely kicking in.

**Comment by arokem** [comment]
In addition, check out functionality added to DIPY in https://github.com/dipy/dipy/pull/2539. Maybe we can use that and not need any additional work on parallelization to happen here?

**Comment by oesteban** [comment]
Closed with #103
